scrapper.py

```diff
@@ -1,10 +1,143 @@
 import requests
 from bs4 import BeautifulSoup
+from sys import argv
+import re
 
-page = requests.get('http://127.0.0.1:5500/krisha_scrapper/flat_list.html')
-soup = BeautifulSoup(page.content, 'html.parser')
+def check_cli_arg(command_line_arguments: list) -> str:
+    """ Function checks:
+            - that URL was entered on command line, if not promts entering URL
+            - URL starts with 'https://krisha.kz/', if not promts entering correct URL or exit
+            and return URL
+    """
 
-prices = soup.select("html body div.cnt-aprts-lst div.apartment div.price")
-first5 = prices[:5]
-for price in first5:
-    print(price.text)
\ No newline at end of file
+    if len(command_line_arguments) == 1:
+        cli_argument = input("Don't input URL. Input URL:")
+    elif len(command_line_arguments) == 2:
+        cli_argument = command_line_arguments[1]
+    elif len(command_line_arguments) > 2:
+        cli_argument = input('Too much arguments. Input URL:')
+
+    while cli_argument.strip()[:18] != "https://krisha.kz/":
+        cli_argument = input('Uncorrect URL. Try again or press ENTER to exit:')
+        if cli_argument == "": exit(0)
+
+    return cli_argument.strip()
+
+def pagescrapper(accepted_url: str, page_number: int) -> str:
+    """ Function scrap site page by url and return html text"""
+
+    url = f'{accepted_url}&page={page_number}'
+    
+    page = requests.get(url, timeout=0.5)
+    if page.status_code < 400:
+        print(f'Page {page_number} scrapped. Status code: {page.status_code}', end = '\n')
+
+    soup = BeautifulSoup(page.content, 'html.parser')
+
+    return soup
+
+def get_number_of_options(soup: str) -> int:
+    """ Function parse scrapped text and return number of options"""
+    
+    text_with_numbers_of_options = soup.find('div',
+                                     class_='a-search-subtitle search-results-nb').text.strip().split()
+    numbers_of_options = int(text_with_numbers_of_options[1]) # number of options
+
+    return numbers_of_options
+
+def get_flat_price(soup: str, prices: list) -> int:
```

Comment:
@ProvVlad , I would suggest to change execution flow to reduce computational complexity of the script, namely, run all 3 functions in 1 cycle which means you invoke page_grabber function and in scope of this function you invoke (1) get_price, (2) get_square, (3) get_id etc.